### PR TITLE
DOP-4908 - Mobile and Tablet versions of universal search bar

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-4908

Mobile and Tablet versions of Universal Search Bar. Please refer to [mobile and Tablet figma designs](https://www.figma.com/design/7cR6eWtFvKTnJIPL8ZsTaB/Content-Template-Projects?node-id=2847-3607&t=h4KJKZ5jY0ly3PZE-0)

### Notable Changes:
- Docs menu is included in action bar in mobile view
- There are separate icons for mobile vs desktop (ie. feedback button)
- Search bar is hidden in mobile view until activated, and right aligned in tablet view
- Should work across all used templates

### Staging Links:

[Landing](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/seung.park/DOP-4908/index.html)
[cloud docs](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4908/index.html)
[docs 404](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/docs-404/seung.park/DOP-4908/index.html)
[changelog template](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4908/reference/api-resources-spec/changelog/index.html)
[testing preview layout](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/seung.park/DOP-4908/cloud-docs/test-wayfinding/index.html)

### Notes:
- 404 page was not mobile friendly (DOP-4957) so made the changes here with the action bar

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: src/components/ActionBar/ActionBar.js [hunk 1]
```diff
@@ -57,13 +37,22 @@ const ActionBar = ({ template, slug, ...props }) => {
...
       {fakeColumns && <div></div>}
       <ActionBarSearchContainer className={cx(searchContainerClassname)}>
-        <SearchInput />
+        {sidenav && (
+          <MobileStyledLink onClick={() => setHideMobile((state) => !state)}>
+            <Icon glyph={hideMobile ? 'ChevronDown' : 'ChevronUp'} />
+            Docs Menu
+          </MobileStyledLink>
+        )}
+        {/*  */}
```
Comment (mmeigs) [inline src/components/ActionBar/ActionBar.js:52]: Remove comment

File: src/components/ActionBar/SearchInput.js [hunk 2]
```diff
@@ -156,31 +171,61 @@ const SearchInput = ({ className }) => {
...
+          onClick={() => setMobileSearchActive(false)}
+        >
+          Cancel
+        </Link>
+      )}
       <SuspenseHelper>
         <Chatbot serverBaseUrl={CHATBOT_SERVER_BASE_URL} darkMode={darkMode}>
           <SearchMenu
-            isOpen={isOpen}
+            isOpen={searchValue.length && isOpen}
```
Comment (mmeigs) [inline src/components/ActionBar/SearchInput.js:208]: nit: Just to match the other instances can you add the double bang?

File: src/components/Header/index.js [hunk 3]
```diff
@@ -17,7 +16,7 @@ const StyledHeaderContainer = styled.header(
   `
 );
 
-const Header = ({ sidenav, eol, template }) => {
+const Header = ({ sidenav, eol, template, slug }) => {
```
Comment (mmeigs) [inline src/components/Header/index.js:19]: Remove if unused?

File: src/components/Header/index.js [hunk 4]
```diff
@@ -49,6 +47,8 @@ const Header = ({ sidenav, eol, template }) => {
 Header.propTypes = {
   sidenav: PropTypes.bool,
   eol: PropTypes.bool.isRequired,
+  slug: PropTypes.string.isRequired,
```
Comment (mmeigs) [inline src/components/Header/index.js:50]: Also, remove if unused?

File: src/components/Sidenav/SidenavMobileMenuDropdown.js [hunk 5]
```diff
@@ -19,16 +21,24 @@ const Container = styled('div')`
...
+  .dark-theme & {
+    color: ${palette.gray.light2};
+  }
 `;
 
-const StyledIcon = styled(Icon)`
-  margin-right: ${theme.size.medium};
+const Text = styled('div')`
+  color: var(--color);
 `;
```
Comment (mmeigs) [inline src/components/Sidenav/SidenavMobileMenuDropdown.js:37]: Would we want to use LG Body? - I do see this is just how it was before you

File: src/theme/docsTheme.js [hunk 6]
```diff
@@ -76,7 +76,7 @@ const header = {
   // used for scrolling elements into place, considering sticky header
   navbarScrollOffset: '175px',
   actionBarHeight: '115px',
-  actionBarMobileHeight: '185px', // TODO: update after mobile designs are finalized
+  actionBarMobileHeight: '60px', // TODO: update after mobile designs are finalized
```
Comment (mmeigs) [inline src/theme/docsTheme.js:79]: Just a reminder :)

File: src/components/ActionBar/ActionBar.js [hunk 7]
```diff
@@ -57,13 +37,22 @@ const ActionBar = ({ template, slug, ...props }) => {
...
+        <SearchInput sidenav={sidenav} />
```
Comment (mmeigs) [inline src/components/ActionBar/ActionBar.js:53]: I don't think you use this prop `sidenav` in SearchInput

File: src/components/ActionBar/ActionBar.js [hunk 8]
```diff
@@ -79,8 +68,9 @@ const ActionBar = ({ template, slug, ...props }) => {
 };
 
 ActionBar.propTypes = {
-  template: PropTypes.string,
-  slug: PropTypes.string,
+  template: PropTypes.string.isRequired,
+  slug: PropTypes.string.isRequired,
+  sidenav: PropTypes.bool.isRequired,
```
Comment (mmeigs) [inline src/components/ActionBar/ActionBar.js:73]: Either make this optional OR add the `sidenav` prop to all ActionBar invocations. It's missing from the `SidenavMobileMenuDropdown` file.

**Review by mmeigs — COMMENTED** [review]
Oh, I am EVER so obsessed with this new feedback button icon. 

Question: in the designs it looks like (either oversight or real) that we are now going to hide the search functionality in the UnifiedHeader. Check with design and then if so, you can pass `hideSearch` to the UnifiedHeader. (I think regardless of size...)

File: src/components/Sidenav/SidenavMobileMenuDropdown.js (hunk 5 above)
Comment (seungpark) [reply src/components/Sidenav/SidenavMobileMenuDropdown.js:37]: oh seems like this is using an Overline in figma. will update!

File: src/components/ActionBar/ActionBar.js (hunk 8 above)
Comment (seungpark) [reply src/components/ActionBar/ActionBar.js:73]: ah idk why i put actionbar in `SidenavMobileMenuDropdown` -> that should be removed! coming up

Comment (seungpark) [reply src/components/ActionBar/ActionBar.js:67]: SidenavMobileMenuDropdown was removed 👍

**Comment by seungpark** [comment]
> Oh, I am EVER so obsessed with this new feedback button icon.
> 
> Question: in the designs it looks like (either oversight or real) that we are now going to hide the search functionality in the UnifiedHeader. Check with design and then if so, you can pass `hideSearch` to the UnifiedHeader. (I think regardless of size...)

Great point! Done! will verify with Design that this is desired (will undo the last commit otherwise)

**Review by mmeigs — APPROVED** [review]
Nice 🚀

**Pull request merged**